ui-src/src/CellMatrix.ts

```diff
@@ -138,36 +145,33 @@ export default class CellMatrix {
     });
   }
 
-  private posToIndex(pos: Pos) {
-    return this.size.x * pos.x + pos.y
-  }
-
-  private xyToIndex(x: number, y: number) {
-    return this.size.x * x + y;
+  private posToIndex(pos: Pos): number {
+    return this.size.x * pos.y + pos.x
   }
 
   private getValue(pos: Pos): number {
     return this.data[this.posToIndex(pos)];
   }
 
   private setValue(pos: Pos, value: number) {
-    this.data[this.size.x * pos.x + pos.y] = value;
+    this.data[this.posToIndex(pos)] = value;
   }
 
   private incrementAdjacentMineCount(pos: Pos) {
-    this.setValue(pos, this.getAdjacentMines(pos)+1);
+    // NB: this can overflow!! use with caution
+    this.data[this.posToIndex(pos)] += 1
```

Comment:
suggest using this.setValue rather than modifying the data directly. But at the same time if it's working maybe we should leave it alone